pkg/services/rpcsrv/server_helper_test.go

```diff
@@ -136,6 +136,9 @@ func wrapUnitTestChain(t testing.TB, chain *core.Blockchain, orc OracleHandler,
 
 	handler := http.HandlerFunc(rpcServer.handleHTTPRequest)
 	srv := httptest.NewServer(handler)
+	t.Cleanup(func() {
+		srv.Close()
```

Comment:
It's not the only place where we should properly close this server. And it's suspicious that we have exactly those tests failing where `src.Close` isn't called. Search through the failing test issues and check whether it solves some of them.